quickwit/quickwit-janitor/src/actors/delete_task_pipeline.rs

```diff
@@ -143,10 +158,7 @@ impl DeleteTaskPipeline {
             SplitsUpdateMailbox::Publisher(publisher_mailbox),
             self.max_concurrent_split_uploads,
         );
-        let (uploader_mailbox, uploader_supervisor_handler) = ctx
-            .spawn_actor()
-            .set_kill_switch(KillSwitch::default())
```

Comment:
The `supervise` function is doing the right thing on this, it calls `kill_switch.child()` and pass the child to the supervised actor.

With this modification, we can now kill the pipeline and expects that all children will be killed too.